packages/squiggle-lang/src/compiler/types.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * An "IR" is an intermediate representation of a Squiggle code.
```

Comment:
I renamed "Expression" to "IR", since "Expression" was also used in context of AST nodes.